lib/pos_blockchain.go

```diff
@@ -1841,7 +1848,19 @@ func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*UtxoView, er
 	if err != nil {
 		return nil, errors.Wrapf(err, "getUtxoViewAtBlockHash: Problem initializing UtxoView")
 	}
+	// TODO: there's another performance enhancement we can make here. If we have a view in the
+	// cache for one of the ancestors, we can skip fetching the block and connecting it by taking
+	// a copy of it and replacing the existing view.
 	for ii := len(uncommittedAncestors) - 1; ii >= 0; ii-- {
+		// For some reason this is slower than just connecting the blocks from badger.
+		//if ancestorView, exists := bc.blockViewCache.Lookup(*uncommittedAncestors[ii].Hash); exists {
+		//	copiedView, err := ancestorView.(*UtxoView).CopyUtxoView()
+		//	if err != nil {
+		//		return nil, errors.Wrapf(err, "getUtxoViewAtBlockHash: Problem copying UtxoView from cache")
+		//	}
+		//	utxoView = copiedView
+		//	continue
+		//}
```

Comment:
Maybe just remove this. Badger is likely to have a lot of memory caching that optimizes lookups in ways that are difficult to match with custom solutions.

Reply:
Well this just copies the view as opposed to getting the block from badger and connecting the block. I figured that copying the view must be cheaper than connecting the block but maybe I'm wrong! Just seems like copying should be faster